src/editor/actions/move_discrete_keyframe_action.cpp

```
@@ -0,0 +1,57 @@
+    , m_finalFrame(finalFrame) {
+}
+
+MoveDiscreteKeyframeAction::~MoveDiscreteKeyframeAction() = default;
+
+void MoveDiscreteKeyframeAction::Do() {
+    auto it = m_entity->m_discreteTracks.find(m_target);
+    if (it == m_entity->m_discreteTracks.end()) {
+        return;
+    }
+    auto& track = it->second;
+    auto* src = track.GetKeyframe(m_initialFrame);
```

Clang (14.0.6)

[error] variable 'src' is not initialized

(cppcoreguidelines-init-variables,-warnings-as-errors)